db/version_set.cc

```diff
@@ -4816,10 +4810,15 @@ std::string ManifestPicker::GetNextManifest(uint64_t* number,
 
 Status VersionSet::TryRecover(
     const std::vector<ColumnFamilyDescriptor>& column_families, bool read_only,
-    std::string* db_id, bool* has_missing_table_file) {
-  ManifestPicker manifest_picker(dbname_, fs_);
-  manifest_picker.SeekToFirstManifest();
-  Status s = manifest_picker.status();
+    std::string* db_id, bool* has_missing_table_file,
+    std::vector<std::string>& files_in_dbname) {
+  Status s;
+  if (read_only) {
+    assert(files_in_dbname.empty());
+    s = fs_->GetChildren(dbname_, IOOptions(), &files_in_dbname,
+                         /*dbg=*/nullptr);
+  }
+  ManifestPicker manifest_picker(dbname_, files_in_dbname, fs_);
```

Comment:
After second thought, let's move this to line 448 of db_impl_open.cc

Reply:
I assume it's within the scope of `if(best_efforts_recovery)`? LIne 448 is in an outer scope in the current commit.